feat(lib): added `stats/incr/nanmmin` package

jalajk3004 opened the pull request:
Resolves #5594

## Description

> What is the purpose of this pull request?

This pull request:

-   adds a package nanmmin, which works like mmin in which it calculates mmin incrementally ignoring NaN values

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5594

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

jalajk3004 committed:
Added benchmark file

jalajk3004 committed:
added the lib

jalajk3004 committed:
added the example

jalajk3004 committed:
added docs

jalajk3004 committed:
added readme and package

jalajk3004 committed:
added tests

jalajk3004 committed:
edit

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

ShabiShett07 commented on lib/node_modules/@stdlib/stats/incr/mmin/lib/main.js:
```diff
@@ -58,94 +58,89 @@ var format = require( '@stdlib/string/format' );
+    }
+}
 
 // EXPORTS //
```
```suggestion
// EXPORTS //

module.exports = incrnanmmin
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmmin/lib/main.js:
```diff
@@ -0,0 +1,151 @@
+* @returns {Function} accumulator function
+*
+* @example
+* var accumulator = incrmmin( 3 );
```
Wrong function name — `incrmmin` should be `incrnanmmin`. Same issue in `docs/types/index.d.ts` line 44 and `README.md` line 46.

```suggestion
* var accumulator = incrnanmmin( 3 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmmin/package.json:
```diff
@@ -0,0 +1,72 @@
+{
+    "name": "@stdlib/stats/incr/nanmmin",
+    "version": "0.0.0",
+    "description": "Compute a moving minimum incrementally, ignoring `NaN` values.",
```
No backticks around NaN in `package.json`. Also, this file uses 4-space indentation — stdlib convention is 2-space for JSON files.

```suggestion
    "description": "Compute a moving minimum incrementally, ignoring NaN values.",
```

Planeshifter committed:
fix: address review feedback in stats/incr/nanmmin

- Delegate to @stdlib/stats/incr/mmin to avoid reimplementing the moving-min logic (mirrors nanmstdev/nanmin), which fixes the PINF sentinel leaking as a result on a leading NaN (now returns null)
- Revert unintended modification of the existing incr/mmin package
- Correct an incorrect expected fixture value in the incremental test
- Fix incrmmin -> incrnanmmin references in README and index.d.ts examples
- Reindent with tabs; remove unused test imports and trailing whitespace
- Normalize package.json formatting; add nan/ignore keywords

Planeshifter reviewed (approved):
LGTM after edits
